### ISSUE TITLE
PixelReplaceStep returns strange output filenames from save_results=True

### ISSUE THREAD

**Issue by stscijgbot-jp** [spacetelescope/jwst#8904]
_Issue [JP-3784](https://jira.stsci.edu/browse/JP-3784) was created on JIRA by [Ned Molter](https://jira.stsci.edu/secure/ViewProfile.jspa?name=emolter):_

Running the PixelReplaceStep using regression test data from nirspec/ifu as follows:

```
fn = glob.glob(stem + "jw01249*_cal.fits")
container = datamodels.ModelContainer(fn)
PixelReplaceStep.call(container, save_results=True, skip=False)
```
returns filenames that look like "step_PixelReplaceStep_3_pixelreplacestep.fits".  The output filenames should be updated to be more descriptive and more closely match other steps.  Each input model in the list is still saved separately according to an index, so setting this fix as low priority.

**Comment by stscijgbot-jp** [comment]
_Comment by [Jane Morrison](https://jira.stsci.edu/secure/ViewProfile.jspa?name=morrison) on [JIRA](https://jira.stsci.edu/browse/JP-3784?focusedCommentId=747851#comment-747851):_

[Ned Molter](https://jira.stsci.edu/secure/ViewProfile.jspa?name=emolter) I looked at the regression tests and I am not seeing where this test might have been. Could you tell me the test you were trying to add the pixel_replacement save results ?

**Comment by stscijgbot-jp** [comment]
_Comment by [Ned Molter](https://jira.stsci.edu/secure/ViewProfile.jspa?name=emolter) on [JIRA](https://jira.stsci.edu/browse/JP-3784?focusedCommentId=757180#comment-757180):_

Hi Jane, this was not itself a regression test.  It was tested using the regression test data in our regtest suite (test_bigdata/jwst-pipeline/dev/nirspec/ifu/jw01249*_cal.fits.

 

You are right that an association would normally be used as input to spec3, but a container is a valid input to an individual step, and indeed containers are passed between steps during the spec3 pipeline.  

 

For other steps, I believe the asn ID in the output filenames is taken from container.asn_table["asn_id"], so for the step run I wrote above it might have been a more representative test if that were assigned manually; however, I don't think that would change the result.

 

It would be good to test what happens here if the individual step is run on an association as you suggest, as well as what you get when you run the entire calwebb_spec3 on an asn with pixel_replace.save_results=True

**Comment by stscijgbot-jp** [comment]
_Comment by [Melanie Clarke](https://jira.stsci.edu/secure/ViewProfile.jspa?name=mclarke) on [JIRA](https://jira.stsci.edu/browse/JP-3784?focusedCommentId=780805#comment-780805):_

Fixed by #9019

Will be in build 11.3, not 11.2.

**Comment by stscijgbot-jp** [comment]
_Comment by [Melanie Clarke](https://jira.stsci.edu/secure/ViewProfile.jspa?name=mclarke) on [JIRA](https://jira.stsci.edu/browse/JP-3784?focusedCommentId=780806#comment-780806):_

Closing - no INS testing needed.
